debugger/src/debug_adapter/dap/adapter.rs

```diff
@@ -822,7 +824,7 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
                             message: Some(format!(
                                 "Source breakpoint at memory address: {address:#010X}"
                             )),
-                            source: None,
+                            source: Some(args.source.clone()),
```

Comment:
This clone should not be required.

Reply:
If I take it off, I get 
```
error[E0382]: borrow of moved value: `args.source`
   --> debugger/src/debug_adapter/dap/adapter.rs:810:25
    |
810 |                         &args.source,
    |                         ^^^^^^^^^^^^ value borrowed here after move
...
827 |                             source: Some(args.source),
    |                                          ----------- value moved here, in previous iteration of loop
    |
    = note: move occurs because `args.source` has type `debug_adapter::dap::dap_types::Source`, which does not implement the `Copy` trait

```